Apps/JobScheduler/src/Jobs/CommunicationJob.cs

```diff
@@ -0,0 +1,167 @@
+
+    /// <inheritdoc />
+    public class CommunicationJob : ICommunicationJob
+    {
+        private const int ConcurrencyTimeout = 30 * 60; // 30 minutes
+        private readonly IConfiguration configuration;
+        private readonly ILogger<CommunicationJob> logger;
+        private readonly ICommunicationDelegate communicationDelegate;
+        private readonly ICommunicationEmailDelegate commEmailDelegate;
+        private readonly IEmailQueueService emailQueueService;
+        private readonly GatewayDbContext dbContext;
+
+        private readonly int retryFetchSize;
+        private readonly string fromEmailAddressHGDonotreply;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CommunicationJob"/> class.
+        /// </summary>
+        /// <param name="configuration">The configuration to use.</param>
+        /// <param name="logger">The logger to use.</param>
+        /// <param name="communicationDelegate">The Communication delegate to use.</param>
+        /// <param name="commEmailDelegate">The Communication Email delegate to use.</param>
+        /// <param name="emailQueueService">The email queue service to use.</param>
+        /// <param name="dbContext">The db context to use.</param>
+        public CommunicationJob(IConfiguration configuration, ILogger<CommunicationJob> logger, ICommunicationDelegate communicationDelegate, ICommunicationEmailDelegate commEmailDelegate, IEmailQueueService emailQueueService, GatewayDbContext dbContext)
+        {
+            Contract.Requires((configuration != null) && (emailQueueService != null));
+            this.configuration = configuration!;
+            this.logger = logger;
+            this.communicationDelegate = communicationDelegate!;
+            this.commEmailDelegate = commEmailDelegate!;
+            this.emailQueueService = emailQueueService!;
+            this.dbContext = dbContext;
+
+            IConfigurationSection emailJobSection = this.configuration.GetSection("EmailJob");
+            this.retryFetchSize = emailJobSection.GetValue<int>("MaxRetryFetchSize", 250);
+            IConfigurationSection commEmailJobSection = this.configuration.GetSection("CreateCommEmailsForNewCommunications");
+            this.fromEmailAddressHGDonotreply = commEmailJobSection.GetValue<string>("FromEmailAddressHGDonotreply");
+        }
+
+        /// <inheritdoc />
+        [DisableConcurrentExecution(ConcurrencyTimeout)]
+        public void CreateCommunicationEmailsForNewCommunications()
+        {
+            this.logger.LogDebug($"Creating emails & communication emails...");
+            List<Communication> communications = this.communicationDelegate.GetEmailCommunicationsInNewProcessingOrError();
+
+            if (communications.Count > 0)
+            {
+                this.logger.LogInformation($"Found {communications.Count} communications which are in New status.");
+
+                foreach (Communication comm in communications)
```

Comment:
nit: comm => communication

Reply:
Will update it, btw personally I prefer using short name for these in the loop, i.e. foreach(Communication c...) or forearch(Blog b ...). This short name helps the code to be easier to read as It is super hard to read "communication" single object vs. "communications" (list)